tests: kernel: intel_adsp / simulator fixes

laurenmurphyx64 opened the pull request:
Miscellaneous fixes for a test known to fail on simulators in general, and some tests specifically on ACE40.

laurenmurphyx64 committed:
tests: smp_suspend: run only on qemu, native and hardware

Control where the test runs.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

laurenmurphyx64 committed:
tests: adjust CONFIG_MAX_THREAD_BYTES for some

For these particular tests, many platforms require higher
CONFIG_MAX_THREAD_BYTES than the default 2, so guard
against failures by setting it high.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

andyross reviewed (commented):
Not really clear to me why that kconfig value (which governs the maximum number of threads at runtime) is specific to the adsp platforms?   Maybe it's just that these are SMP boards and have more idle threads?  But if these are hitting it surely everyone else is very close too.  Might be cleaner just to bump it for everyone.

laurenmurphyx64 commented:
> Not really clear to me why that kconfig value (which governs the maximum number of threads at runtime) is specific to the adsp platforms? Maybe it's just that these are SMP boards and have more idle threads? But if these are hitting it surely everyone else is very close too. Might be cleaner just to bump it for everyone.

I wouldn't mind that. Daniel introduced the overlays for PTL, though, so he might have more to say on this. What do you think @dcpleung ?

dcpleung commented:
We could set this to an arbitrary big number as default in the test. Though the issue is that this will overwrite the board default config, where some boards have defaults as high as 6. Maybe set it to 8 and call it a day for now?

laurenmurphyx64 commented:
@dcpleung @andyross I set the default to 8 :+1: There was another test (tests/lib/c_lib/thrd/prj.conf) hitting this on ACE30 (but not on ACE40 for some reason?) that had an overlays for ACE30 upping it to 3 - I changed the default to 8 there too, and deleted the overlays.
